encoding/protobuf/protoc-gen-yarpc-go/internal/lib/lib.go

```diff
@@ -642,6 +661,43 @@ func clientServerStreamingMethods(service *protoplugin.Service) ([]*protoplugin.
 	return methods, nil
 }
 
+func fileDescriptorClosureVarName(f *protoplugin.File) (string, error) {
+	name := f.GetName()
+	if name == "" {
+		return "", fmt.Errorf("Could not create fileDescriptorClosureVarName: %s has no name", f)
+	}
+	h := sha256.Sum256([]byte(name))
+	return fmt.Sprintf("yarpcFileDescriptorClosure%s", hex.EncodeToString(h[:8])), nil
+}
+
+func encodedFileDescriptor(f *protoplugin.File) (string, error) {
+	b, err := f.SerializedFileDescriptor()
+	if err != nil {
+		return "", err
+	}
+
+	buf := make([]byte, 0, 0)
+	w := bytes.NewBuffer(buf)
+	w.WriteString("[]byte{\n")
```

Comment:
The `WriteString` method indicates that using a [`strings.Builder`](https://golang.org/pkg/strings/#Builder) is more efficient.

This is probably pretty minor however since we do all of this work before the server is even set up

Reply:
Changed this but realised `strings.Builder` is a `1.10` api. Staying with `bytes.Buffer` for now.